endpoint: Do not fail to regenerate endpoint when running w/o L7 proxy

brb opened the pull request:
It is expected that in some deployments the L7 proxy will be disabled. Currently, it can be disabled when `--install-iptables-rules=false` is set which can happen when e.g.  running in the ipvlan mode or
w/o netfilter/iptables.

To play nicely with such deployments, do not fail the endpoint regeneration when the L7 proxy is disabled.

Fixes #9518

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9481)
<!-- Reviewable:end -->

brb committed:
endpoint: Do not fail to regenerate w/o L7 proxy

It is expected that in some deployments the L7 proxy will be disabled.
Currently, it can be disabled when `--install-iptables-rules=false`
is set which can happen when e.g.  running in the ipvlan mode or
w/o netfilter/iptables.

To play nicely with such deployments, do not fail the endpoint
regeneration when the L7 proxy is disabled.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

ianvernon reviewed (changes requested):
Commented out of band of the review.

brb committed:
daemon: Disable L7 proxy with explicit flag

Previously, the L7 proxy was disabled if "--install-iptables-rules"
was set to false which might be not clear for a user.

Instead, add "--enable-l7-proxy" (defaults to true) flag to control
whether the proxy should be disabled or not.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

joestringer reviewed (approved):
The changes that are here are simple and straightforward.

I understand from the community meeting that there is a desire to also reject CNPs that configure L7 rules (and perhaps also FQDN rules) in this PR as well.

brb committed:
daemon: Enable FQDN proxy if --enable-l7-proxy is set

Previously, the proxy was disabled if `--install-iptables-rules`
was set to false, which could have been not obvious for a user.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

brb committed:
helm: Add global.l7Proxy.enabled param

The param controls whether the L7 proxy is enabled (defaults to true).

Signed-off-by: Martynas Pumputis <m@lambda.lt>

brb committed:
docs: Fix ipvlan iptables-free gsg

Disable L7 proxy, as otherwise cilium will panic due to
"--install-iptables-rules" set to false.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

aanm commented on install/kubernetes/cilium/charts/config/templates/configmap.yaml:
```diff
@@ -193,6 +193,12 @@ data:
 {{- end }}
 
+{{- if .Values.global.l7Proxy }}
+
```
remove new line?

aanm committed:
pkg/policy: show error if user installs a L7 CNP with L7 proxy disabled

A cilium node with a L7 proxy enabled will show the following status
in the node status:

```
status:
  nodes:
    k8s1:
      enforcing: true
      lastUpdated: "2019-10-29T14:09:49.449575508Z"
      localPolicyRevision: 2
      ok: true
```

A cilium node without a L7 proxy enabled will show the following status
in the node status:

```
status:
  nodes:
    k8s1:
      error: 'Invalid CiliumNetworkPolicy spec: L7 policy is not supported since L7
        proxy is not enabled'
      lastUpdated: "2019-10-29T14:12:23.309604539Z"
```

Signed-off-by: André Martins <andre@cilium.io>

joestringer commented on daemon/daemon.go:
```diff
@@ -433,11 +433,11 @@ func NewDaemon(ctx context.Context, dp datapath.Datapath) (*Daemon, *endpointRes
 			option.Config.AccessLog, &d, option.Config.AgentLabels, d.datapath, d.endpointManager)
 	} else {
-		log.Warning("L7 proxies not supported when --install-iptables-rules=\"false\"")
+		log.Info("L7 proxies is disabled")
```
nit:
```suggestion
		log.Info("L7 proxies are disabled")
```

joestringer reviewed (approved):
Minor nit.
